platform/backend/src/mcp-server-runtime/manager.ts

```diff
@@ -40,36 +113,47 @@ export class McpServerRuntimeManager {
   constructor() {
     this.k8sConfig = new k8s.KubeConfig();
 
-    // Load K8s config from environment or default locations
+    const hasCustomKubeconfig =
```

Comment:
Declaring `hasCustomKubeconfig` makes us do non-null assertions `kubeconfig!` later in the code 2 times. Perhaps we shouldn't declare `hasCustomKubeconfig` and just throw if the config is set but empty?